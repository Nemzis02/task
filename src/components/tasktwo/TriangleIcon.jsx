import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  position: relative;
  width: 25px;
  height: 25px;
`;

const IconImage = styled.img`
  position: absolute;
  left: 0;
  top: 0;
  user-select: none;
  width: 100%;
  height: 100%;
`;

const Triangle = styled.div`
  position: absolute;
  top: 1px;
  left: 1px;
  width: 0;
  height: 0;
  border-top: 0;
  border-bottom: 21px solid #ca2128;
  border-left: 11px solid transparent;
  border-right: 12px solid transparent;
`;

const TriangleIcon = () => (
  <Container>
    <Triangle />
    <IconImage src="images/triangle.svg" alt="Triangle" />
  </Container>
);

export default TriangleIcon;
