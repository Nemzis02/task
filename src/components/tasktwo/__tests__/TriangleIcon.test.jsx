import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TriangleIcon from '../TriangleIcon';

describe('TriangleIcon Component', () => {
  it('Should be rendered correctly', () => {
    const { container } = render(<TriangleIcon />);
    expect(container).toMatchSnapshot();
  });
});
