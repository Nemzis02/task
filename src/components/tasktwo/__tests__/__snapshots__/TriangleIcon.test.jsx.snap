// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriangleIcon Component Should be rendered correctly 1`] = `
<div>
  <div
    class="TriangleIcon__Container-kelki1-0 dBbNcD"
  >
    <div
      class="TriangleIcon__Triangle-kelki1-2 iABfzB"
    />
    <img
      alt="Triangle"
      class="TriangleIcon__IconImage-kelki1-1 fcbLiJ"
      src="images/triangle.svg"
    />
  </div>
</div>
`;
